meta-resin-common/recipes-support/resin-state-reset/resin-state-reset/resin-state-reset

```diff
@@ -9,8 +9,10 @@ populate() {
     # Re-construct directory structure and populate files
     if [ -f $2 ]; then
         mkdir -p "$(dirname $1)" && cp $2 $1
+    elif [ -d "$2" ]; then
+        mkdir -p -- "$1" && find "$2" \( -type f -o -type l \) -exec cp {} "$1" +
     else
-        mkdir -p $1 && find $2 \( -type f -o -type l \) -exec cp {} $1 +
+        mkdir -p -- "$1"
```

Comment:
So this gets created if $2 is neither a file or a directory? What case is that and how did you encounter it? It seems to me that if we end up in that state something is off